### PR TITLE
fix: resolve issues around `InlineEditor` [DET-7914]

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
- [DET-7914]
-  Cannot ignore keydown event until the IME is confirmed
- Description in uncategorized table has no limit but the one in experiment detail has 500 letters limit
- `isDarkOn` is deprecated in `InlineEditor` 

## Test Plan
- Check if `InlineEditor` works as extected
- Check if description in uncategorized table has 500 letters limit
- Check if display name in settings has 32 letters limit
- Check if display name is ellipt when its too long
- Check if IME ignores keydown event until the IME is confirmed

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-7914]: https://determinedai.atlassian.net/browse/DET-7914?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/InlineEditor.test.tsx [hunk 1]
```diff
@@ -89,9 +89,28 @@ describe('InlineEditor', () => {
...
+    const textbox = screen.getByRole('textbox');
+    await user.click(textbox);
+    await user.clear(textbox);
+    await user.type(textbox, 'こんにちは');
+    fireEvent.keyDown(textbox, {
+      charCode: 229,
+      code: 'Enter',
+      key: 'Enter',
+      keyCode: 229,
+    });
```
Comment (keita-determined) [inline webui/react/src/components/InlineEditor.test.tsx:109]: I dont wanna use `fireEvent` here but seems like userEvent doesnt have option to specify keyCode

File: webui/react/src/components/InlineEditor.tsx [hunk 2]
```diff
@@ -114,20 +111,21 @@ const InlineEditor: React.FC<Props> = ({
       e.preventDefault();
       return;
     }
-    if (e.code === CODE_ENTER) {
-      if (!allowNewline || !e.shiftKey) e.preventDefault();
+    if (e.which === 229) {
+      // Ignore keydown event until IME is confirmed like Japanese and Chinese
+      return;
+    }
```
Comment (hkang1) [inline webui/react/src/components/InlineEditor.tsx:117]: ah interesting, learned something new today!

File: webui/react/src/components/InlineEditor.test.tsx (hunk 1 above)
Comment (hkang1) [reply webui/react/src/components/InlineEditor.test.tsx]: yeah, totally get it. I have encountered something like this where fireEvent still had to be used

**Review by hkang1 — APPROVED** [review]
Thanks for catching this issue and fixing it! Not to mention testing it via IME keyboard too.

**Pull request merged**